newIDE/app/src/SceneEditor/index.js

```diff
@@ -154,6 +154,10 @@ type State = {|
 
   // State for tags of objects:
   selectedObjectTags: SelectedTags,
+
+  renamedObjectWithContext: ?ObjectWithContext,
+  setRenamedObjectWithContext: ?(obj: ObjectWithContext) => void,
```

Comment:
I don't think you need this state variable. It's a props of `ObjectsList` that you set with a method of `SceneEditor`

Reply:
Hi @AlexandreSi ,

Sorry, I think I need it because I have this method used in ToolbarCommand:

```typescript
_startRenamingObject = () => {
  this._onEditRenameObject(this.state.selectedObjectWithContext);
};
```

`_onEditRenameObject` sets this state:

```typescript
this.setState(
  {
	renamedObjectWithContext: objectWithContext
  }
);
```

Setting the state `renamedObjectWithContext` activates the input in the Objects List so the user can write the new name.

If we leave the state only in the ObjectList, I can't see how I can change in from the SceneEditor/index.js. That was on of my big problems.

If you have any clue about this tell me and I will check it.

Reply:
Good one there. Done and pushed.